.github/workflows/haskell.yml

```diff
@@ -123,6 +123,11 @@ jobs:
         skip: "${{ vars.BINARY_CACHE_URI != '' }}"
         enable-save: false
 
+    - name: Build core components
+      continue-on-error: true
+      run: |
+        cabal build cardano-node cardano-cli cardano-node-chairman cardano-submit-api -j1
+
```

Comment:
Why do you need this new step?

Reply:
It's because of this: https://github.com/haskell/cabal/issues/9587

Reply:
Ideally we need a `retries: 1` option on the step, bit Github Actions doesn't support such a thing.

Reply:
Can you add a link to https://github.com/haskell/cabal/issues/9587 in comment? So that we know why is there, and can potentially remove it in the future when https://github.com/haskell/cabal/issues/9587 is solved.

Reply:
Done!